src/drawers/drawer.ts

```diff
@@ -60,6 +74,37 @@ export class Drawer {
   }
 
   /**
+   * Retieves a function that can supply initial settings to entering elements.
+   * this function is typically supplied by the plot using the Drawer
+   */
+  public initializer(): () => AttributeToProjector;
+  /**
+   * Sets the initializer function for the Drawer.
+   * This function returns an AttributeToProjector that is applied
+   * to the new elements appended to the enter() selection
+   * Typically set from _createDrawer in the plot
+   *
+   * @param {() => AttributeToProjector} the function.
+   * @returns {Drawer} The calling Drawer.
+   */
+  public initializer(fnattrToProjector: () => AttributeToProjector): Drawer;
+  public initializer(fnattrToProjector?: () => AttributeToProjector): any {
+    if (fnattrToProjector == null) {
+      return this._initializer;
+    }
+    this._initializer = fnattrToProjector;
+    return this;
+  }
```

Comment:
Because doing it at the plot level can't distinguish between entering and updated elements - so what happens now is that existing elements get overridden with initial state. This refactor applies this initial state only to entering elements:

```javascript
    this._drawingTarget.enter = dataElements.enter()
      .append(this._svgElementName)
      .attr(this.appliedInitializer());
```